### ISSUE TITLE
Podspec-defined dependency fails while same podfile-defined dependency succeeds.

### ISSUE THREAD

**Issue by crow** [CocoaPods/CocoaPods#5128]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

Cococapods version 1.0.0.beta6

# Report

## What did you do?

Added Dependencies two ways, outlined in Case 1 and Case 2:

Case 1: Added a pod dependency to my podfile - and ensured that the dependency is not present in my podspec:

Podfile contents:
```ruby
source 'https://github.com/CocoaPods/Specs.git'

target 'UrbanAirship-iOS-GA-Tracker_Example' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'UrbanAirship-iOS-SDK','~> 7.0.3.rc1' # this is the dependency
end

target 'UrbanAirship-iOS-GA-Tracker_Tests' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'OCMock'
end
```

Case 2: Added my pod dependency to my podspec.json - and removed the pod dependency from the podfile: 

Podspec contents:

  "dependencies" : {
    "UrbanAirship-iOS-SDK": ["~> 7.0.3.rc1"] // this is the dependency
  },

Podfile contents::
```ruby
source 'https://github.com/CocoaPods/Specs.git'

target 'UrbanAirship-iOS-GA-Tracker_Example' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
end

target 'UrbanAirship-iOS-GA-Tracker_Tests' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'OCMock'
end
```
## What did you expected to happen?
I expected the behavior to be the same between Case 1 and Case 2 (see previous section).

## What happened instead?

Actual results case 1: Pod installs successfully:

> Analyzing dependencies
> Fetching podspec for `UrbanAirship-iOS-GA-Tracker` from `..`
> Downloading dependencies
> Installing UrbanAirship-iOS-GA-Tracker (1.0.0)
> Installing UrbanAirship-iOS-SDK (7.0.3.rc1)
> Generating Pods project
> Integrating client project

[!] Please close any current Xcode sessions and use `UrbanAirship-iOS-GA-Tracker.xcworkspace` for this project from now on.

Actual results case 2: Pod fails to install with error: 

> [!] Unable to satisfy the following requirements:
> 
> - `UrbanAirship-iOS-SDK (~> 7.0.3.rc1)` required by `UrbanAirship-iOS-GA-Tracker (1.0.0)`
> 
> It seems like you've changed the constraints of dependency `UrbanAirship-iOS-SDK` inside your development pod `UrbanAirship-iOS-GA-Tracker`.
> You should run `pod update UrbanAirship-iOS-SDK` to apply changes you've made.

## Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'

target 'UrbanAirship-iOS-GA-Tracker_Example' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'UrbanAirship-iOS-SDK','~> 7.0.3.rc1'
end

target 'UrbanAirship-iOS-GA-Tracker_Tests' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'OCMock'
end
```

**Comment by orta** [comment]
Can you try with this Podfile please, this won't work on 0.39

```ruby
target 'UrbanAirship-iOS-GA-Tracker_Example' do
  pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
  pod 'UrbanAirship-iOS-SDK','~> 7.0.3.rc1'

  target 'UrbanAirship-iOS-GA-Tracker_Tests' do
    inherit! :search_paths
    pod 'OCMock'
  end
end

```

**Comment by segiddins** [comment]
@crow  thanks for reporting this! could you share the entirety of the podspec so we can try and get this to resolve locally? Thanks!

**Comment by crow** [comment]
podfile contents:

```
{
  "name": "UrbanAirship-iOS-GA-Tracker",
  "documentation_url": "https://github.com/urbanairship/ios-ga-tracker",
  "version": "1.0.0",
  "license": "BSD",
  "summary": "A simple way to integrate Urban Airship and Google Analytics.",
  "homepage": "https://github.com/urbanairship/ios-ga-tracker",
  "authors": {
    "Urban Airship": "support@urbanairship.com"
  },
  "source": {
    "git": "https://github.com/urbanairship/ios-ga-tracker.git",
    "tag": "1.0.0"
  },
  "requires_arc": true,
  "dependencies" : {
    "GoogleAnalytics": ["~> 3.0"],
    "UrbanAirship-iOS-SDK": ["~> 7.0.3.rc1"]
  },
  "source_files": [
      "Pod/Classes/*"
  ]
}

```

**Comment by crow** [comment]
the suggested podfile contents work:

> 
> Can you try with this Podfile please, this won't work on 0.39
> 
> target 'UrbanAirship-iOS-GA-Tracker_Example' do
>   pod 'UrbanAirship-iOS-GA-Tracker',:path => ".."
>   pod 'UrbanAirship-iOS-SDK','~> 7.0.3.rc1'
> 
>   target 'UrbanAirship-iOS-GA-Tracker_Tests' do
>     inherit! :search_paths
>     pod 'OCMock'
>   end
> end

**Comment by crow** [comment]
Why would one ever see the following error:

> It seems like you've changed the constraints of dependency `Analytics` inside your development pod `UrbanAirship-iOS-Segment-Extension`.
> You should run `pod update Analytics` to apply changes you've made.

After running `pod deintegrate`? 

Where else is state being saved?

**Comment by neonichu** [comment]
Looks like this was resolved with 1.0 beta, so I'm going to close it.

**Comment by anton-matosov** [comment]
I trapped into this issue with 1.0.1 GA release and the only thing that helped is to run `pod setup`.
I had to put it as part of my Travis-CI before-install script, otherwise it was constantly failing.
